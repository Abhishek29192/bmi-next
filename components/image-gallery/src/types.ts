import { AcceptedNode } from "@bmi/media";

export type Image = {
  media?: React.ReactElement<AcceptedNode>;
  thumbnail?: string;
  /**
   * @deprecated The media itself should have the alternative text.
   */
  altText?: string;
  caption?: string;
};
