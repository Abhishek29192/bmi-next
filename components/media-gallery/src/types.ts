import React from "react";
import { AcceptedNode } from "@bmi/media";

export type Media = {
  media?: React.ReactElement<AcceptedNode>;
  thumbnail?: string;
  caption?: string;
  altText?: string;
  isVideo?: boolean;
};
