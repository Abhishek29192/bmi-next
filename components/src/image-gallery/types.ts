import { AcceptedNode } from "../media/Media";

export type Image = {
  media?: React.ReactElement<AcceptedNode>;
  thumbnail?: string;
  /**
   * @deprecated The media itself should have the alternative text.
   */
  altText?: string;
  caption?: string;
};
