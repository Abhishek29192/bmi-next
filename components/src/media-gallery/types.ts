import React from "react";
import { AcceptedNode } from "../media/Media";

export type Media = {
  media?: React.ReactElement<AcceptedNode>;
  thumbnail?: string;
  caption?: string;
  altText?: string;
  isVideo?: boolean;
  visualiserParameters?: Record<string, string | number>;
  openVisualiser?: (params?: Record<string, unknown>) => void;
};
