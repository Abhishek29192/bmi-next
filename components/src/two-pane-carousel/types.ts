import { SVGImport } from "@bmi-digital/svg-import";
import { ClickableAction } from "../clickable/Clickable";
import { AcceptedNode } from "../media/Media";

export type Slide = {
  brandIcon?: SVGImport;
  title: React.ReactNode;
  description?: React.ReactNode;
  cta?: {
    label: React.ReactNode;
    action?: ClickableAction;
  };
  media?: React.ReactElement<AcceptedNode>;
};

export type Props = {
  slides: Slide[];
};
