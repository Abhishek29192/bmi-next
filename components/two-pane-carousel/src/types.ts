import { ClickableAction } from "@bmi/anchor-link";
import { AcceptedNode } from "@bmi/media";
import { SVGImport } from "@bmi/svg-import";

export type Slide = {
  brandIcon?: SVGImport;
  title: React.ReactNode;
  description?: React.ReactNode;
  cta?: {
    label: React.ReactNode;
    action?: ClickableAction;
  };
  media?: React.ReactElement<AcceptedNode>;
};

export type Props = {
  slides: Slide[];
};
