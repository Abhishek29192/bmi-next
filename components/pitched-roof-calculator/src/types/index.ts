export type RangeValue = {
  start: number;
  end: number;
  value: number;
};

export type BaseProduct = {
  code: string;
  name: string;
};

export type BaseVariant = BaseProduct & {
  externalProductCode: string;
  image: string;
};

export type LengthBasedProduct = BaseVariant & {
  length: number;
};

export type WidthBasedProduct = BaseVariant & {
  width: number;
};

export type VergeMetalFlushOption = {
  type: "METAL_FLUSH";
  name: string;
  left: LengthBasedProduct;
  right: LengthBasedProduct;
  leftStart?: LengthBasedProduct;
  rightStart?: LengthBasedProduct;
};

export type VergeTileOption = {
  type: "TILE";
  name: string;
  left: WidthBasedProduct;
  right: WidthBasedProduct;
  halfLeft: WidthBasedProduct;
  halfRight: WidthBasedProduct;
};

export type VergeOption = VergeTileOption | VergeMetalFlushOption;

export type Accessory = BaseVariant & {
  category: ProductCategory;
  packSize?: number;
};

export type Underlay = BaseVariant & {
  minSupportedPitch: number;
  length: number;
  width: number;
  overlap: number;
};

export type GutteringVaraint = LengthBasedProduct & {
  downpipe: Accessory;
  downpipeConnector: Accessory;
};

export type Guttering = BaseProduct & {
  image: string;
  variants: GutteringVaraint[];
};

export type BaseTile = {
  minBattenGauge: number;
  maxBattenGauge: RangeValue[];
  eaveGauge: RangeValue[];
  ridgeSpacing: RangeValue[];
  width: number;
  height: number;
  brokenBond: boolean;
};

export type MainTileVariant = BaseVariant &
  BaseTile & {
    color: string;
    halfTile: WidthBasedProduct;
    hip: LengthBasedProduct;
    ridgeOptions: LengthBasedProduct[];
    vergeOptions: VergeOption[];
    valleyMetalFlushStart: LengthBasedProduct;
    valleyMetalFlush: LengthBasedProduct;
    valleyMetalFlushEnd: LengthBasedProduct;
    valleyMetalFlushTop: LengthBasedProduct;
    valleyMetalFlushDormerStart: LengthBasedProduct;
    accessories: Accessory[];
    eaveAccessories: Accessory[];
    clip: Accessory;
    ridgeAndHipScrew: Accessory;
    longScrew: Accessory;
    screw: Accessory;
    stormBracket: Accessory;
    finishingKit: Accessory;
    ventilationHoodOptions: Accessory[];
  };

export type MainTileCategory = "concrete" | "metal" | "clay";

export type MainTile = BaseProduct &
  BaseTile & {
    category: MainTileCategory;
    variants: MainTileVariant[];
  };

export type ProductCategory =
  | "tiles"
  | "fixings"
  | "sealing"
  | "ventilation"
  | "accessories";

export type ResultsRow = {
  category: ProductCategory;
  image: string;
  description: string;
  externalProductCode: string;
  packSize: string;
  quantity: number;
};

export type ResultsObject = {
  [category in ProductCategory]: ResultsRow[];
};
