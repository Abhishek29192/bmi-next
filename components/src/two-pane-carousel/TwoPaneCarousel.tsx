import React from "react";
import { useMediaQuery } from "@material-ui/core";
import { useTheme } from "@material-ui/core";
import { Props } from "./types";
import styles from "./TwoPaneCarousel.module.scss";
import MobileTwoPaneCarousel from "./_MobileTwoPaneCarousel";
import TabletTwoPaneCarousel from "./_TabletTwoPaneCarousel";

const TwoPaneCarousel = ({
  children,
  ...props
}: Props & { children?: React.ReactNode }) => {
  const theme = useTheme();
  const matches = useMediaQuery(theme.breakpoints.up("md"));

  const carousel = matches ? (
    <TabletTwoPaneCarousel {...props} />
  ) : (
    <MobileTwoPaneCarousel {...props} />
  );

  return (
    <div className={styles["TwoPaneCarousel"]}>
      {carousel}
      {children && <div className={styles["wrapper"]}>{children}</div>}
    </div>
  );
};

export default TwoPaneCarousel;
