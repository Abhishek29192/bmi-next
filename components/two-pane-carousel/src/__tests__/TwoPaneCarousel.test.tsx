import React from "react";
import mediaQuery from "css-mediaquery";
import { render, cleanup, fireEvent } from "@testing-library/react";
import snapshotDiff from "snapshot-diff";
import imageSource from "test-image.jpg";
import BrandIcon from "brand-logo.svg";
import mockConsole from "jest-mock-console";
import TwoPaneCarousel from "../";
import { Props } from "../types";

beforeAll(() => {
  mockConsole();
});

const createMatchMedia = (width: number) => {
  return (query: string): MediaQueryList =>
    ({
      matches: mediaQuery.match(query, { width }),
      addListener: () => {},
      removeListener: () => {}
    } as unknown as MediaQueryList);
};

const slides: Props["slides"] = [
  {
    brandIcon: BrandIcon,
    title: "H1 Heading running onto 2 lines",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ligula nisi, condimentum facilisis hendrerit eget, sollicitudin non sapien. Class aptent taciti sociosqu ad litora.",
    cta: {
      label: "Read the full story"
    },
    media: <img src={imageSource} alt="Lorem ipsum" />
  },
  {
    title: "H1 Heading",
    description:
      "Aliquip velit exercitation sunt eiusmod. Ipsum est quis dolore cupidatat nisi reprehenderit aliquip exercitation. Magna mollit Lorem est aliqua consequat officia cillum dolor.",
    media: <img src={imageSource} alt="Lorem ipsum" />
  },
  {
    title: "H1 Heading",
    description: "Aliquip velit exercitation sunt eiusmod.",
    cta: {
      label: "Read the full story"
    },
    // TODO: This tests the deprecated imageSource
    imageSource
  },
  {
    title: "H1 Heading",
    media: <img src={imageSource} alt="Lorem ipsum" />
  }
];

afterEach(cleanup);
describe("TwoPaneCarousel component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <TwoPaneCarousel slides={slides}>Lorem ipsum</TwoPaneCarousel>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a single pane carousel", () => {
    window.matchMedia = createMatchMedia(1280);

    const { container } = render(
      <TwoPaneCarousel slides={slides}>Lorem ipsum</TwoPaneCarousel>
    );
    expect(container).toMatchSnapshot();
  });
  it("navigates to next page", () => {
    const nextLabel = "next";
    const { container, getByLabelText } = render(
      <TwoPaneCarousel slides={slides}>Lorem ipsum</TwoPaneCarousel>
    );
    const containerBeforeClick = container!.cloneNode(true);

    fireEvent.click(getByLabelText(nextLabel));

    expect(snapshotDiff(containerBeforeClick, container)).toMatchSnapshot();
  });
  it("navigates to previous page", () => {
    window.matchMedia = createMatchMedia(1280);
    const previousLabel = "previous";
    const { container, getByLabelText } = render(
      <TwoPaneCarousel slides={slides}>Lorem ipsum</TwoPaneCarousel>
    );
    const containerBeforeClick = container.cloneNode(true);

    fireEvent.click(getByLabelText(previousLabel));

    expect(snapshotDiff(containerBeforeClick, container)).toMatchSnapshot();
  });
  it("navigates to next page on mobile", () => {
    window.matchMedia = createMatchMedia(480);
    const nextLabel = "next";
    const { container, getByLabelText } = render(
      <TwoPaneCarousel slides={slides}>Lorem ipsum</TwoPaneCarousel>
    );
    const containerBeforeClick = container.cloneNode(true);

    fireEvent.click(getByLabelText(nextLabel));

    expect(snapshotDiff(containerBeforeClick, container)).toMatchSnapshot();
  });
  it("navigates to previous page on mobile", () => {
    window.matchMedia = createMatchMedia(480);
    const previousLabel = "previous";
    const { container, getByLabelText } = render(
      <TwoPaneCarousel slides={slides}>Lorem ipsum</TwoPaneCarousel>
    );
    const containerBeforeClick = container!.cloneNode(true);

    fireEvent.click(getByLabelText(previousLabel));

    expect(snapshotDiff(containerBeforeClick, container)).toMatchSnapshot();
  });
});
