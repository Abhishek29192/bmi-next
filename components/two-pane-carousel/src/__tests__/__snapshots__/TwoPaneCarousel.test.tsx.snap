// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoPaneCarousel component navigates to next page 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -19,11 +19,11 @@
              <div
                style=\\"overflow-x: hidden;\\"
              >
                <div
                  class=\\"react-swipeable-view-container\\"
-                 style=\\"flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;\\"
+                 style=\\"flex-direction: row; transition: transform 0.35s cubic-bezier(0.15, 0.3, 0.25, 1) 0s; direction: ltr; display: flex; will-change: transform; transform: translate(-400%, 0);\\"
                >
                  <div
                    aria-hidden=\\"false\\"
                    class=\\"Carousel__slide--global\\"
                    data-swipeable=\\"true\\"
@@ -34,38 +34,20 @@
                    >
                      <div
                        class=\\"slide left-pane-slide\\"
                        style=\\"width: 100%;\\"
                      >
-                       <svg
-                         class=\\"brand-icon\\"
-                         preserveAspectRatio=\\"xMinYMin\\"
-                       />
                        <h2
                          class=\\"MuiTypography-root heading Typography Typography--underline MuiTypography-h1\\"
                        >
-                         H1 Heading running onto 2 lines
+                         H1 Heading
                        </h2>
                        <div
                          class=\\"text\\"
-                       >
-                         Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ligula nisi, condimentum facilisis hendrerit eget, sollicitudin non sapien. Class aptent taciti sociosqu ad litora.
-                         <div>
-                           <a
-                             class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover cta Anchorlink MuiTypography-colorPrimary\\"
-                             type=\\"button\\"
                        >
-                             <span>
-                               Read the full story
-                             </span>
-                             <svg
-                               aria-hidden=\\"true\\"
-                               class=\\"icon\\"
-                               focusable=\\"false\\"
-                             />
-                           </a>
-                         </div>
+                         Aliquip velit exercitation sunt eiusmod. Ipsum est quis dolore cupidatat nisi reprehenderit aliquip exercitation. Magna mollit Lorem est aliqua consequat officia cillum dolor.
+                         <div />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
@@ -104,16 +86,21 @@
            >
              <span
                class=\\"sliding-slot\\"
              >
                <span
-                 class=\\"numbers\\"
+                 class=\\"numbers numbers--up\\"
+               >
+                 <span
+                   class=\\"number\\"
                  >
+                   01
+                 </span>
                  <span
                    class=\\"number number--active\\"
                  >
-                   01
+                   02
                  </span>
                </span>
              </span>
              <span
                class=\\"total\\"
@@ -160,11 +147,11 @@
            <div
              style=\\"overflow-x: hidden;\\"
            >
              <div
                class=\\"react-swipeable-view-container\\"
-               style=\\"flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;\\"
+               style=\\"flex-direction: row; transition: transform 0.35s cubic-bezier(0.15, 0.3, 0.25, 1) 0s; direction: ltr; display: flex; will-change: transform; transform: translate(-400%, 0);\\"
              >
                <div
                  aria-hidden=\\"false\\"
                  class=\\"Carousel__slide--global\\"
                  data-swipeable=\\"true\\""
`;

exports[`TwoPaneCarousel component navigates to previous page 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -19,11 +19,11 @@
              <div
                style=\\"overflow-x: hidden;\\"
              >
                <div
                  class=\\"react-swipeable-view-container\\"
-                 style=\\"flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;\\"
+                 style=\\"flex-direction: row; transition: transform 0.35s cubic-bezier(0.15, 0.3, 0.25, 1) 0s; direction: ltr; display: flex; will-change: transform; transform: translate(-200%, 0);\\"
                >
                  <div
                    aria-hidden=\\"false\\"
                    class=\\"Carousel__slide--global\\"
                    data-swipeable=\\"true\\"
@@ -34,23 +34,19 @@
                    >
                      <div
                        class=\\"slide left-pane-slide\\"
                        style=\\"width: 100%;\\"
                      >
-                       <svg
-                         class=\\"brand-icon\\"
-                         preserveAspectRatio=\\"xMinYMin\\"
-                       />
                        <h2
                          class=\\"MuiTypography-root heading Typography Typography--underline MuiTypography-h1\\"
                        >
-                         H1 Heading running onto 2 lines
+                         H1 Heading
                        </h2>
                        <div
                          class=\\"text\\"
                        >
-                         Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ligula nisi, condimentum facilisis hendrerit eget, sollicitudin non sapien. Class aptent taciti sociosqu ad litora.
+                         Aliquip velit exercitation sunt eiusmod.
                          <div>
                            <a
                              class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover cta Anchorlink MuiTypography-colorPrimary\\"
                              type=\\"button\\"
                            >
@@ -104,14 +100,19 @@
            >
              <span
                class=\\"sliding-slot\\"
              >
                <span
-                 class=\\"numbers\\"
+                 class=\\"numbers numbers--down\\"
                >
                  <span
                    class=\\"number number--active\\"
+                 >
+                   03
+                 </span>
+                 <span
+                   class=\\"number\\"
                  >
                    01
                  </span>
                </span>
              </span>
@@ -160,11 +161,11 @@
            <div
              style=\\"overflow-x: hidden;\\"
            >
              <div
                class=\\"react-swipeable-view-container\\"
-               style=\\"flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;\\"
+               style=\\"flex-direction: row; transition: transform 0.35s cubic-bezier(0.15, 0.3, 0.25, 1) 0s; direction: ltr; display: flex; will-change: transform; transform: translate(-200%, 0);\\"
              >
                <div
                  aria-hidden=\\"false\\"
                  class=\\"Carousel__slide--global\\"
                  data-swipeable=\\"true\\"
@@ -175,14 +176,13 @@
                  >
                    <div
                      class=\\"slide\\"
                      style=\\"width: 100%;\\"
                    >
-                     <img
-                       alt=\\"Lorem ipsum\\"
-                       class=\\"image Media\\"
-                       src=\\"path-to-image\\"
+                     <div
+                       class=\\"image\\"
+                       style=\\"background-image: url(path-to-image);\\"
                      />
                    </div>
                  </div>
                </div>
              </div>"
`;

exports[`TwoPaneCarousel component renders a single pane carousel 1`] = `
<div
  class="TwoPaneCarousel"
>
  <div
    class="MuiGrid-root Grid MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-flex-start"
  >
    <div
      class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-xl-4"
    >
      <div
        class="left-pane"
      >
        <div
          class="wrapper"
        >
          <div
            class="Carousel Carousel--opacity Carousel--swipable"
          >
            <div
              style="overflow-x: hidden;"
            >
              <div
                class="react-swipeable-view-container"
                style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
              >
                <div
                  aria-hidden="false"
                  class="Carousel__slide--global"
                  data-swipeable="true"
                  style="width: 100%; flex-shrink: 0; overflow: auto;"
                >
                  <div
                    class="page"
                  >
                    <div
                      class="slide left-pane-slide"
                      style="width: 100%;"
                    >
                      <svg
                        class="brand-icon"
                        preserveAspectRatio="xMinYMin"
                      />
                      <h2
                        class="MuiTypography-root heading Typography Typography--underline MuiTypography-h1"
                      >
                        H1 Heading running onto 2 lines
                      </h2>
                      <div
                        class="text"
                      >
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ligula nisi, condimentum facilisis hendrerit eget, sollicitudin non sapien. Class aptent taciti sociosqu ad litora.
                        <div>
                          <a
                            class="MuiTypography-root MuiLink-root MuiLink-underlineHover cta Anchorlink MuiTypography-colorPrimary"
                            type="button"
                          >
                            <span>
                              Read the full story
                            </span>
                            <svg
                              aria-hidden="true"
                              class="icon"
                              focusable="false"
                            />
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="SlideControls controls"
        >
          <button
            aria-label="previous"
            class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root chevron"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <span
            class="middle-container"
          >
            <span
              class="sliding-slot"
            >
              <span
                class="numbers"
              >
                <span
                  class="number number--active"
                >
                  01
                </span>
              </span>
            </span>
            <span
              class="total"
            >
              03
            </span>
          </span>
          <button
            aria-label="next"
            class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root chevron"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-xl-8"
    >
      <div
        class="wrapper"
      >
        <div
          class="Carousel Carousel--swipable"
        >
          <div
            style="overflow-x: hidden;"
          >
            <div
              class="react-swipeable-view-container"
              style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
            >
              <div
                aria-hidden="false"
                class="Carousel__slide--global"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  class="page"
                >
                  <div
                    class="slide"
                    style="width: 100%;"
                  >
                    <img
                      alt="Lorem ipsum"
                      class="image Media"
                      src="path-to-image"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="wrapper"
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`TwoPaneCarousel component renders correctly 1`] = `
<div
  class="TwoPaneCarousel"
>
  <div>
    <div
      class="wrapper"
    >
      <div
        class="Carousel Carousel--opacity Carousel--swipable"
      >
        <div
          style="overflow-x: hidden;"
        >
          <div
            class="react-swipeable-view-container"
            style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
          >
            <div
              aria-hidden="false"
              class="Carousel__slide--global"
              data-swipeable="true"
              style="width: 100%; flex-shrink: 0; overflow: auto;"
            >
              <div
                class="page"
              >
                <div
                  class="slide left-pane-slide"
                  style="width: 100%;"
                >
                  <svg
                    class="brand-icon"
                    preserveAspectRatio="xMinYMin"
                  />
                  <h2
                    class="MuiTypography-root heading Typography Typography--underline MuiTypography-h2"
                  >
                    H1 Heading running onto 2 lines
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="Carousel Carousel--opacity Carousel--swipable"
      >
        <div
          style="overflow-x: hidden;"
        >
          <div
            class="react-swipeable-view-container"
            style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
          >
            <div
              aria-hidden="false"
              class="Carousel__slide--global"
              data-swipeable="true"
              style="width: 100%; flex-shrink: 0; overflow: auto;"
            >
              <div
                class="page"
              >
                <div
                  class="slide left-pane-slide"
                  style="width: 100%;"
                >
                  <img
                    alt="Lorem ipsum"
                    class="image Media"
                    src="path-to-image"
                  />
                  <div
                    class="text"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ligula nisi, condimentum facilisis hendrerit eget, sollicitudin non sapien. Class aptent taciti sociosqu ad litora.
                    <div>
                      <a
                        class="MuiTypography-root MuiLink-root MuiLink-underlineHover cta Anchorlink MuiTypography-colorPrimary"
                        type="button"
                      >
                        <span>
                          Read the full story
                        </span>
                        <svg
                          aria-hidden="true"
                          class="icon"
                          focusable="false"
                        />
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="SlideControls SlideControls--full-size controls"
    >
      <button
        aria-label="previous"
        class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root chevron"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <span
        class="middle-container"
      >
        <span
          class="sliding-slot"
        >
          <span
            class="numbers"
          >
            <span
              class="number number--active"
            >
              01
            </span>
          </span>
        </span>
        <span
          class="total"
        >
          03
        </span>
      </span>
      <button
        aria-label="next"
        class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root chevron"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="wrapper"
  >
    Lorem ipsum
  </div>
</div>
`;
